app/plugin.go

```diff
@@ -886,6 +890,44 @@ func (a *App) processPrepackagedPlugin(pluginPath *pluginSignaturePath) (*plugin
 	return plugin, nil
 }
 
+// installFeatureFlagPlugins handles the automatic installation/upgrade of plugins from feature flags
+func (a *App) installFeatureFlagPlugins() {
+	ffControledPlugins := a.Config().FeatureFlags.Plugins()
+
+	// Respect the automatic prepackaged disable setting
+	if !*a.Config().PluginSettings.AutomaticPrepackagedPlugins {
+		return
+	}
+
+	for pluginId, version := range ffControledPlugins {
+		// Skip installing if the plugin has been previously disabled.
+		pluginState := a.Config().PluginSettings.PluginStates[pluginId]
+		if pluginState != nil && !pluginState.Enable {
+			a.Log().Debug("Not auto installing/upgrade because plugin was disabled", mlog.String("plugin_id", pluginId), mlog.String("version", version))
+			continue
+		}
+
+		// Check if we already installed this version as InstallMarketplacePlugin can't handle re-installs well.
+		pluginStatus, err := a.Srv().GetPluginStatus(pluginId)
+		if err == nil && pluginStatus.Version == version {
+			continue
+		}
+
+		if version != "" && version != "control" {
```

Comment:
What is `control` in this context?

Reply:
`control` is what split.io will return if the feature flag does not exist.